plugins/common/forward/src/index.ts

```diff
@@ -1,21 +1,42 @@
-import { Command, Context, Dict, Schema, segment, Session, Time } from 'koishi'
 import { parsePlatform } from '@koishijs/helpers'
+import { Context, Dict, Schema, segment, Session, Time } from 'koishi'
 
-declare module 'koishi' {
-  interface Channel {
-    forward: string[]
-  }
+export interface Filter {
+  type: 'user' | 'flag' | 'all'
+  data?: string[]
 }
 
 export interface Rule {
   source: string
+  filter: Filter
   target: string
   selfId: string
   guildId?: string
 }
 
+export interface DBRule {
+  id: number
+  source: string
+  filter: Filter
+  target: string
+}
+
+declare module 'koishi' {
+  interface Tables {
+    forward: DBRule
+  }
+}
+
+// @ts-ignore
```

Comment:
If `@ts-ignore` appears in your code, it's probably because you wrote something wrong.